src/comm_manager.cpp

```diff
@@ -237,36 +237,46 @@ void CommManager::offboard_control_callback(const CommLinkInterface::OffboardCon
 {
   // put values into a new command struct
   control_t new_offboard_command;
-  new_offboard_command.x.value = control.x.value;
-  new_offboard_command.y.value = control.y.value;
-  new_offboard_command.z.value = control.z.value;
-  new_offboard_command.F.value = control.F.value;
+  new_offboard_command.Qx.value = control.Qx.value;
+  new_offboard_command.Qy.value = control.Qy.value;
+  new_offboard_command.Qz.value = control.Qz.value;
+  new_offboard_command.Fx.value = control.Fx.value;
+  new_offboard_command.Fy.value = control.Fy.value;
+  new_offboard_command.Fz.value = control.Fz.value;
 
   // Move flags into standard message
-  new_offboard_command.x.active = control.x.valid;
-  new_offboard_command.y.active = control.y.valid;
-  new_offboard_command.z.active = control.z.valid;
-  new_offboard_command.F.active = control.F.valid;
+  new_offboard_command.Qx.active = control.Qx.valid;
+  new_offboard_command.Qy.active = control.Qy.valid;
+  new_offboard_command.Qz.active = control.Qz.valid;
+  new_offboard_command.Fx.active = control.Fx.valid;
+  new_offboard_command.Fy.active = control.Fy.valid;
+  new_offboard_command.Fz.active = control.Fz.valid;
 
   // translate modes into standard message
   switch (control.mode) {
     case CommLinkInterface::OffboardControl::Mode::PASS_THROUGH:
-      new_offboard_command.x.type = PASSTHROUGH;
-      new_offboard_command.y.type = PASSTHROUGH;
-      new_offboard_command.z.type = PASSTHROUGH;
-      new_offboard_command.F.type = THROTTLE;
+      new_offboard_command.Qx.type = PASSTHROUGH;
+      new_offboard_command.Qy.type = PASSTHROUGH;
+      new_offboard_command.Qz.type = PASSTHROUGH;
+      new_offboard_command.Fx.type = PASSTHROUGH;
+      new_offboard_command.Fy.type = PASSTHROUGH;
+      new_offboard_command.Fz.type = PASSTHROUGH;
```

Comment:
Does the input from ROSplane need to change to accommodate this?

Reply:
No, other than renaming the command from "F" to "Fx". This just means it won't be scaled by the `max_thrust` and `max_throttle` parameters, which are really designed for multirotors anyway.

Also note that the fixedwing aircraft mixer doesn't take in forces and moments, really. It takes in delta commands, which correspond one to one with servo and motor setpoints. This means that the mixers for fixedwings are fundamentally different than for multirotors, since they don't need to be inverted, won't use motor parameters, etc. It's set up this way to avoid loading all the aerodynamic coefficients into the firmware.